specs/carousel/carousel.md

```diff
@@ -0,0 +1,258 @@
+
+
+### Risks and Challenges
+
+*Notable risks or challenges associated with implementing the component. Would we need to make any breaking changes in order to achieve this component's goals?*
+
+### Prior Art/Examples
+- [FAST-DNA (React)](https://explore.fast.design/components/carousel)
+- [W3C](https://w3c.github.io/aria-practices/examples/carousel/carousel-1.html)
+
+---
+
+## Design
+
+### API
+
+*Component name:*
+- `fast-carousel`
+
+*Attributes:*
+- `autoplay` - boolean value to toggle if the carousel plays on load or not (defaults to false)
+- `autoplay-interval` - a number representing the autoplay interval in milliseconds
+- `loop` - boolean value to toggle if the carousel loops slides at the beginning or end (defaults to true)
+- `paused` - whether the rotation is paused or not
+- `aria-roledescription` - set to *carousel* for overall component, and set to *slide* for each slide
+- `aria-labelledby` - optional based on implementation**
+- `aria-label` - optional, based on implementation**
+
+** See the [W3C Specification](https://w3c.github.io/aria-practices/#wai-aria-roles-states-and-properties-4) for requirements and details.
+
+*CSS custom properties:*
+- ???
+
+*Slots:*
+- `default` - slot for slides content // TODO: IS this the right way?
+- `play-toggle` - the control to toggle if the slides rotate or not
+- `next-flipper` - the control used to rotate to the next slide
+- `previous-flipper` - the control used to rotate to the previous slide
+- `tablist` - the control to select a certain slide
+
+*Events*
+- `on-pause` - callback fired when the rotation is stopped, such as when keyboard focus or hover is on any element of the carousel, or the *play-toggle* is set to off.
+- `on-active-id-update` - callback fired when the active id is changed
+
+### Anatomy and Appearance
+**Carousel Structure:**
+
+Note: the Carousel class will have a method that checks the attribute names and grabs the items.
+
+```html
+<template>
+    <div
+        class="carousel"
+        aria-roledescription="carousel"
+        tabindex="-1"
+    >
+    <slot name="play-toggle"> SEE DEFAULT </slot>
+    <slot name="next-flipper"> SEE DEFAULT </slot>
+    <div
+        class="carousel-items"
+        aria-live="off"
+    >
+        <!-- Solution 1 -->
+        <!-- Map all items and use css class's to display correct item based on id. -->
+        <!-- May be a less performant way as the items will be mapped each time. -->
+        ${ x => x.items.map(item, index => {
+            return `
+            <div
+                class="carousel-item ${x => x.getActiveIndex() === index ? 'active' : ''}"
+                role="group"
+                aria-roledescription="slide"
+                aria-label="${x => x.getActiveIndex()} of ${x => x.getItemCount()}"
+            >
+                ${item}
+            </div>`
+            }).join('')
+        }
+        <!-- Solution 2 -->
```

Comment:
Only having one item in the DOM might make slide transitions difficult or impossible to achieve.

Reply:
Yes, a typical carousel has a few so I'll look at updating this.

Reply:
Updated Solution 2 to have 3 slides